infra/gcp/lib.sh

```diff
@@ -434,3 +434,29 @@ function ensure_dns_zone() {
         --dns-name "${dns_name}"
   fi
 }
+
+# Allow a Kubernetes service account (KSA) the ability to authenticate as the as
+# the given GCP service account for the given GKE Project's Kubernetes
+# namespace; this is also called Workload Identity.
+#
+# $1: The GKE project wherein the KSA resides; used to construct the KSA.
+# $2: The K8s namespace; used to construct the KSA.
+# $3: The GCP project that owns the GCP service account; used to construct the KSA.
+# $4: The GCP service account to draw powers from
+function empower_ksa_to_svcacct() {
+    if [ ! $# -eq 4 -o -z "$1" -o -z "$2" -o -z "$3" -o -z "$4" ]; then
+        echo "empower_ksa_to_svcacct(gke_project, k8s_namespace, gcp_project, gcp_scvacct) requires 4 arguments" >&2
+        return 1
+    fi
+
+    gke_project="$1"
+    k8s_namespace="$2"
+    gcp_project="$3"
+    gcp_svcacct="$4"
+
+    gcloud iam service-accounts add-iam-policy-binding \
+        --role roles/iam.workloadIdentityUser \
+        --member "serviceAccount:${gke_project}.svc.id.goog[${k8s_namespace}/${gcp_project}]" \
```

Comment:
Rather than construct the IDNS this way, I'd rather we pass it in fully formed.  This ensures that, should that format ever change or extend, we don't need to change this function or duplicate it or anything.

Also I think the assumption that there's one SA per project is dubious - why couldn't we have 2 GSAs in k8s-artifacts-prod that were each used by different prow jobs?  This conflates the name of the KSA in the GKE cluster with the name of the project the GSA exists in.  It's fine to name them the same, but don't assume they are ALWAYS named the same.

IOW, change args 2 and 3 into a single string, with the documented example "${gke_project}.svc.id.goog[${k8s_namespace}/${k8s_svcacct}]"

Reply:
An [existing example](https://github.com/kubernetes/test-infra/blob/c5d80629ff8a2eb577d05b3dd88986ff2e5a0c48/prow/cluster/build_serviceaccounts.yaml#L17-L18
) where the service account name in the GKE cluster (`k8s-kops-test`) does not match the GCP project name (`kubernetes-jenkins-pull`).

An [existing example](https://github.com/kubernetes/test-infra/blob/c5d80629ff8a2eb577d05b3dd88986ff2e5a0c48/prow/cluster/trusted_serviceaccounts.yaml#L6-L14) where we are using mulitple GCP service accounts (`resultstore`, `pusher`, `deployer`) in the same project (`k8s-prow`).

Reply:
> Also I think the assumption that there's one SA per project is dubious - why couldn't we have 2 GSAs in k8s-artifacts-prod that were each used by different prow jobs? This conflates the name of the KSA in the GKE cluster with the name of the project the GSA exists in. It's fine to name them the same, but don't assume they are ALWAYS named the same.

As I was writing the code, I felt that this assumption was on shaky ground. Thanks for the guidance!